src/Features/Core/Portable/ExtractInterface/AbstractExtractInterfaceService.cs

```diff
@@ -147,50 +164,129 @@ public ExtractInterfaceResult ExtractInterfaceFromAnalyzedType(ExtractInterfaceT
                 solutionWithFormattedInterfaceDocument,
-                documentIds,
+                symbolMapping.DocumentIds,
                 refactoringResult.DocumentToExtractFrom.Id,
-                typeNodeSyntaxAnnotation,
+                symbolMapping.TypeNodeAnnotation,
                 refactoringResult.TypeToExtractFrom,
                 extractedInterfaceSymbol,
                 extractInterfaceOptions.IncludedMembers,
-                symbolToDeclarationAnnotationMap,
-                cancellationToken);
+                symbolMapping.SymbolToDeclarationAnnotationMap,
+                cancellationToken).ConfigureAwait(false);
 
             return new ExtractInterfaceResult(
                 succeeded: true,
                 updatedSolution: completedSolution,
                 navigationDocumentId: interfaceDocumentId);
         }
 
-        private Dictionary<ISymbol, SyntaxAnnotation> CreateSymbolToDeclarationAnnotationMap(
-            IEnumerable<ISymbol> includedMembers,
-            ref Solution solution,
-            out List<DocumentId> documentIds,
-            SyntaxNode typeNode,
-            out SyntaxAnnotation typeNodeAnnotation,
-            CancellationToken cancellationToken)
+        private async Task<ExtractInterfaceResult> ExtractInterfaceToSameFileAsync(
+            Solution solution, ExtractInterfaceTypeAnalysisResult refactoringResult, INamedTypeSymbol extractedInterfaceSymbol,
+            ExtractInterfaceOptionsResult extractInterfaceOptions, CancellationToken cancellationToken)
+        {
+            // Track all of the symbols we need to modify, which includes the original type declaration being modified
+            var symbolMapping = await CreateSymbolToDeclarationAnnotationMapAsync(
+                extractInterfaceOptions.IncludedMembers,
+                solution,
+                refactoringResult.TypeNode,
+                cancellationToken).ConfigureAwait(false);
+
+            var document = symbolMapping.AnnotatedSolution.GetDocument(refactoringResult.DocumentToExtractFrom.Id);
+            var originalRoot = document.GetSyntaxRootSynchronously(cancellationToken);
+            var typeDeclaration = originalRoot.GetAnnotatedNodes(symbolMapping.TypeNodeAnnotation).Single();
+
+            var trackedDocument = document.WithSyntaxRoot(originalRoot.TrackNodes(typeDeclaration));
+
+            var currentRoot = trackedDocument.GetSyntaxRootSynchronously(cancellationToken);
+            var editor = new SyntaxEditor(currentRoot, symbolMapping.AnnotatedSolution.Workspace);
+
+            // Generate the interface syntax node, which will be inserted above the type it's extracted from
+            var codeGenService = trackedDocument.GetLanguageService<ICodeGenerationService>();
+            var interfaceNode = codeGenService.CreateNamedTypeDeclaration(extractedInterfaceSymbol).WithAdditionalAnnotations(SimplificationHelpers.SimplifyModuleNameAnnotation);
```

Comment:
1. wrap.
2. what's this annotation for?

Reply:
So we don't run the simplifier on all code, just marked code that we generated. It's used in `GetFormattedSolutionAsync`